cromwell-backend/src/main/scala/cromwell/backend/Backend.scala

```diff
@@ -0,0 +1,39 @@
+package cromwell.backend
+
+import cromwell.backend.model.Subscription
+import cromwell.caching.Cacheable
+
+/**
+  * Defines basic functionality to interact with a Backend.
+  */
+trait Backend extends Cacheable {
+  /**
+    * Prepare the task and context for execution.
+    */
+  def prepare(): Unit
+
+  /**
+    * Executes task in given context.
+    */
+  def execute(): Unit
+
+  /**
+    * Stops a task execution.
+    */
+  def stop(): Unit
+
+  /**
+    * Performs a cleanUp after the task was executed.
+    */
+  def cleanUp(): Unit
+
+  /**
+    * Subscribe to events on backend.
+    */
+  def subscribeToEvent[T](subscription: Subscription[T]): Unit
+
+  /**
+    * Unsubscribe to events on backend.
+    */
+  def unsubscribeToEvent[T](subscription: Subscription[T]): Unit
+}
```

Comment:
We do it by hand - partially because no one wanted to put one together and partially because there's a lot of subjectivity in scala. Personally I don't like those things because I feel people overly rely on the automated checking when a lot of things aren't easily codified.

We do have a (https://gist.github.com/geoffjentry/17e8baeb03c2678a10fa)[style guide] - which I just realized wasn't publicly listed on the cromwell site. You'll note we're not 100% in compliance ourselves, but we try.